manifest-build-tools/lib/manifest.py

```diff
@@ -0,0 +1,247 @@
+        self.gitbit.set_identity(config.gitbit_identity['username'], config.gitbit_identity['email'])
+        if self._git_credentials:
+            for url_cred_pair in self._git_credentials:
+                url, cred = url_cred_pair.split(',')
+                self.gitbit.add_credential_from_variable(url, cred)
+
+
+    def read_manifest_file(self, filename):
+        """
+        Reads the manifest file json data to class member _manifest.
+        :param filename: where to read the manifest data from
+        :return: None
+        """
+        if not os.path.isfile(filename):
+            raise KeyError("No file found for manifest at {0}".format(filename))
+
+        with open(filename, "r") as manifest_file:
+            self._manifest = json.load(manifest_file)
+
+    def parse_manifest(self):
+        """
+        parse manifest and assign properties
+        """
+
+        if 'build-name' in self._manifest:
+            self._build_name = self._manifest['build-name']
+  
+        if 'build-requirements' in self._manifest:
+            self._build_requirements = self._manifest['build-requirements']
+
+        if 'repositories' in self._manifest:     
+            for repo in self._manifest['repositories']:
+                self._repositories.append(repo)
+
+        if 'downstream-jobs' in self._manifest:
+            for job in self._manifest['downstream-jobs']:
+                self._downstream_jobs.append(job)
+
+
+    @staticmethod
+    def validate_repositories(repositories):
+        """
+        validate whether the entry 'repositories' contains useful information
+        return: True if the entry is valid,
+                False if the entry is unusable
+                and message including omission imformation
+        """
+        result = True
+        message = []
+        for repo in repositories:
+            valid = True
+            # repository url is required
+            if 'repository' not in repo:
+                valid = False
+                message.append("entry without tag repository")
+
+            # either branch or commit-id should be set.
+            if 'branch' not in repo and \
+               'commit-id' not in repo:
+                valid = False
+                message.append("Either branch or commit-id should be set for entry")
+
+            if not valid:
+                result = False
+                message.append("entry content:")
+                message.append("{0}".format(json.dumps(repo, indent=True)))
+        
+        return result, message
+
+    @staticmethod
+    def validate_downstream_jobs(downstream_jobs):
+        """
+        validate whether the entry 'downstream-jobs' contains useful information
+        return: True if the entry is valid,
+                False if the entry is unusable
+                and message including omission imformation
+        """
+        result = True
+        message = []
+        for job in downstream_jobs:
+            valid = True
```

Comment:
Why use result and valid? valid is better than result.
The same problem exists in validate_* methods.